src/components/walletconnect/WalletConnectScan.js

```diff
@@ -0,0 +1,267 @@
+// @flow
+/* eslint-disable*/
+import React, { useCallback, useMemo, useEffect, useState } from 'react'
+import { Picker, ScrollView, View } from 'react-native'
+import { t } from '@lingui/macro'
+import { sortBy } from 'lodash'
+// components
+import Wrapper from '../common/layout/Wrapper'
+import { CustomButton, Section, Image } from '../common'
+import InputText from '../common/form/InputText'
+import InputWithAdornment from '../common/form/InputWithAdornment'
+
+import { withStyles } from '../../lib/styles'
+import { isMobile } from '../../lib/utils/platform'
+import QrReader from '../dashboard/QR/QRScanner'
+import QRCameraPermissionDialog from '../dashboard/SendRecieveQRCameraPermissionDialog'
+import Icon from '../common/view/Icon'
+
+// hooks
+import usePermissions from '../permissions/hooks/usePermissions'
+import useCameraSupport from '../browserSupport/hooks/useCameraSupport'
+import { useClipboardPaste } from '../../lib/hooks/useClipboard'
+
+// utils
+import logger from '../../lib/logger/js-logger'
+import { decorate, ExceptionCategory, ExceptionCode } from '../../lib/exceptions/utils'
+import { useDialog } from '../../lib/dialog/useDialog'
+import normalize from '../../lib/utils/normalizeText'
+import { readWalletConnectUri, useWalletConnectSession, useChainsList } from '../../lib/wallet/WalletConnectClient'
+import { WcHeader } from './WalletConnectModals'
+
+import wcExample from '../../assets/walletconnectExample.png'
+
+const log = logger.child({ from: 'WalletConnectScan' })
+
+const TITLE = `WalletConnect`
+
+const QR_DEFAULT_DELAY = 300
+
+type WalletConnectProps = {
+  styles: {},
+  theme: {},
+  screenProps: any,
+}
+
+const Divider = ({ size = 50 }) => <Section.Separator color="transparent" width={size} style={{ zIndex: -10 }} />
+
+const SwitchChain = ({ switchChain, chainId }) => {
```

Comment:
i suggest to move SwtichChain, ConnectedState, PasteCode and ScanCode  to the ./components
could be placed in a single file

Reply:
done